### PR TITLE
409 - "Sobelow" added to project and CI

### PR DESCRIPTION
**Pull request by Samkeer1**
Added Sobelow to project and CI to check for security vulnerabilities on push.

### CODE REVIEW

**Review by quinnmccourt — APPROVED** [review]

File: lib/slax/projects/project_repos.ex [hunk 1]
```diff
@@ -42,6 +42,7 @@ defmodule Slax.ProjectRepos do
     |> Repo.transaction()
   end
 
+  # sobelow_skip ["DOS.BinToAtom"]
```
Comment (grossvogel) [inline lib/slax/projects/project_repos.ex:45]: There probably isn't a serious vulnerability here, but it's always nice to consider the "skip" comment as a last resort. In this case, that second argument in `Multi.update()` [can actually be any elixir term](https://hexdocs.pm/ecto/Ecto.Multi.html#t:name/0), so making it an atom is completely unnecessary. We could just use `"project_repo_#{id}"` without the colon

File: mix.exs [hunk 2]
```diff
@@ -46,6 +46,7 @@ defmodule Slax.Mixfile do
       {:mox, "~> 1.1", only: :test},
       {:jason, "~> 1.1"},
       {:ex_machina, "~> 2.2", only: :test},
+      {:sobelow, "~> 0.13"},
```
Comment (grossvogel) [inline mix.exs:49]: A not-super-important optimization here is to only install sobelow in test and dev environments, and not have it available at runtime. The Sobelow docs suggest this:
```
{:sobelow, "~> 0.13", only: [:dev, :test], runtime: false}
```

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -42,6 +42,9 @@ jobs:
           mix local.rebar --force
           mix deps.get
 
+      - name: Security Check
+        run: mix sobelow --private --skip --exit
+
       - name: Run Tests
         env:
           MIX_ENV: test
```
Comment (grossvogel) [inline .github/workflows/test.yml:50]: Another optimization we should adopt as a pattern is to hoist this `MIX_ENV=test` up to the top of the whole workflow. If you run different steps with different `MIX_ENV` settings, you can end up compiling the whole app more than once, which makes every workflow run take more time. You can see that if you expand the `Security Check` and the `Run Tests` sections of this run: https://github.com/revelrylabs/slax/actions/runs/9180952793/job/25246558083

**Review by grossvogel — COMMENTED** [review]
I know we're going to be updating several repos, so I'm being ignoring and adding some nits to an already-merged PR. The hope is to develop the best pattern we can before duplicating it 😈

**Pull request merged**